osv/models.py

```diff
@@ -364,7 +364,25 @@ def _tokenize(self, value):
       return []
 
     value_lower = value.lower()
-    return re.split(r'\W+', value_lower) + [value_lower]
+
+    # Deconstructs the id given into parts by retrieving parts that are
+    # alphanumeric.
+    tokens = {token for token in re.split(r'\W+', value_lower) if token}
+    tokens.add(value_lower)
+
+    # Add subsection combinations from id (split at '-') in the search indices
+    # Specifically addresses situation in which UBUNTU-CVE-XXXs don't show up
+    # when searching for the CVE-XXX.
+    # Does not account for combinations with the suffix sections ':' like SUSE
+    parts = value_lower.split('-')
+    num_parts = len(parts)
+    if num_parts > 1:
+      for length in range(2, num_parts + 1):
+        for i in range(num_parts - length + 1):
+          sub_parts = parts[i:i + length]
```

Comment:
nit: personally, I would find this easier to follow:
```python
for i in range(num_parts-1):
  for j in range(i+1, num_parts):
    sub_parts = parts[i:j+1]
```

(I think that's equivalent?)